controllers/common.go

```diff
@@ -13,12 +34,404 @@ func getServiceAccountName() string {
+	}
+	switch deploymentName {
+	case apiDeploymentName:
+
+		if instance.Spec.Deployments.API.ReplicaCount != 0 {
+			count = instance.Spec.Deployments.API.ReplicaCount
+		}
+		return &count
+	case collectorDeploymentName:
+		return &count
+	case indexerDeploymentName:
+		return &count
+	case postgresDeploymentName:
+		return &count
+	}
+	log.V(2).Info("Unknown deployment %s ", deploymentName)
+	return &count
+
+}
+func getImageSha(deploymentName string, instance *searchv1alpha1.Search) string {
+	switch deploymentName {
+	case apiDeploymentName:
+		if instance.Spec.Deployments.API.ImageOverride != "" {
+			return instance.Spec.Deployments.API.ImageOverride
+		}
+		return os.Getenv("API_IMAGE")
+	case collectorDeploymentName:
+		if instance.Spec.Deployments.Collector.ImageOverride != "" {
+			return instance.Spec.Deployments.Collector.ImageOverride
+		}
+		return os.Getenv("COLLECTOR_IMAGE")
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.ImageOverride != "" {
+			return instance.Spec.Deployments.Indexer.ImageOverride
+		}
+		return os.Getenv("INDEXER_IMAGE")
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.ImageOverride != "" {
+			return instance.Spec.Deployments.Database.ImageOverride
+		}
+		return os.Getenv("POSTGRES_IMAGE")
+	}
+	log.V(2).Info("Unknown deployment %s ", deploymentName)
+	return ""
+}
+
+func hasDeployments(instance *searchv1alpha1.Search) bool {
+	return instance.Spec.Deployments.DeepCopy() != nil
+
+}
+
+func isDeploymentCustomized(deploymentName string, instance *searchv1alpha1.Search) bool {
+	if !hasDeployments(instance) {
+		return false
+	}
+	switch deploymentName {
+	case apiDeploymentName:
+		if instance.Spec.Deployments.API.DeepCopy() != nil {
+			return true
+		}
+	case collectorDeploymentName:
+		if instance.Spec.Deployments.Collector.DeepCopy() != nil {
+			return true
+		}
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.DeepCopy() != nil {
+			return true
+		}
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.DeepCopy() != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func isResourcesCustomized(deploymentName string, instance *searchv1alpha1.Search) bool {
+	if !isDeploymentCustomized(deploymentName, instance) {
+		return false
+	}
+	switch deploymentName {
+	case apiDeploymentName:
+		if instance.Spec.Deployments.API.Resources != nil {
+			return true
+		}
+	case collectorDeploymentName:
+		if instance.Spec.Deployments.Collector.Resources != nil {
+			return true
+		}
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.Resources != nil {
+			return true
+		}
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.Resources != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *SearchReconciler) createOrUpdateConfigMap(ctx context.Context, cm *corev1.ConfigMap) (*reconcile.Result, error) {
+	found := &corev1.ConfigMap{}
+	err := r.Get(context.TODO(), types.NamespacedName{
```

Comment:
Use the context that gets passed to the function `ctx`.  There's several cases below.

Reply:
Fixed it.